crates/router/src/core/unified_connector_service.rs

```diff
@@ -0,0 +1,265 @@
+            );
+            metadata.append(
+                consts::UCS_HEADER_API_KEY,
+                parse_metadata_value(api_key.peek(), consts::UCS_HEADER_API_KEY)?,
+            );
+            metadata.append(
+                consts::UCS_HEADER_KEY1,
+                parse_metadata_value(key1.peek(), consts::UCS_HEADER_KEY1)?,
+            );
+            metadata.append(
+                consts::UCS_HEADER_API_SECRET,
+                parse_metadata_value(api_secret.peek(), consts::UCS_HEADER_API_SECRET)?,
+            );
+        }
+        ConnectorAuthType::BodyKey { api_key, key1 } => {
+            metadata.append(
+                consts::UCS_HEADER_AUTH_TYPE,
+                parse_metadata_value(consts::UCS_AUTH_BODY_KEY, consts::UCS_HEADER_AUTH_TYPE)?,
+            );
+            metadata.append(
+                consts::UCS_HEADER_API_KEY,
+                parse_metadata_value(api_key.peek(), consts::UCS_HEADER_API_KEY)?,
+            );
+            metadata.append(
+                consts::UCS_HEADER_KEY1,
+                parse_metadata_value(key1.peek(), consts::UCS_HEADER_KEY1)?,
+            );
+        }
+        ConnectorAuthType::HeaderKey { api_key } => {
+            metadata.append(
+                consts::UCS_HEADER_AUTH_TYPE,
+                parse_metadata_value(consts::UCS_AUTH_HEADER_KEY, consts::UCS_HEADER_AUTH_TYPE)?,
+            );
+            metadata.append(
+                consts::UCS_HEADER_API_KEY,
+                parse_metadata_value(api_key.peek(), consts::UCS_HEADER_API_KEY)?,
+            );
+        }
+        _ => {
+            return Err(UnifiedConnectorServiceError::FailedToObtainAuthType)
+                .attach_printable("Unsupported ConnectorAuthType for header injection")?;
+        }
+    }
+
+    Ok(metadata)
+}
+
+pub fn handle_unified_connector_service_response_for_payment_authorize(
+    response: PaymentsAuthorizeResponse,
+) -> CustomResult<
+    (AttemptStatus, Result<PaymentsResponseData, ErrorResponse>),
+    UnifiedConnectorServiceError,
+> {
+    let status = AttemptStatus::foreign_try_from(response.status())?;
+
+    let router_data_response = match status {
+        AttemptStatus::Charged |
+        AttemptStatus::Authorized |
+        AttemptStatus::AuthenticationPending |
+        AttemptStatus::DeviceDataCollectionPending => Ok(PaymentsResponseData::TransactionResponse {
+            resource_id: hyperswitch_domain_models::router_request_types::ResponseId::ConnectorTransactionId(response.connector_response_reference_id().to_owned()),
+            redirection_data: Box::new(None),
+            mandate_reference: Box::new(None),
+            connector_metadata: None,
+            network_txn_id: None,
+            connector_response_reference_id: Some(response.connector_response_reference_id().to_owned()),
+            incremental_authorization_allowed: None,
+            charges: None,
+        }),
+        _ => Err(ErrorResponse {
```

Comment:
We can't hard-code other status as 500, only the http error can be 500, All status needs to be handled properly